### PR TITLE
Fixed #24105: Pass the path returned by upload_to through the get_valid_name of storage

### PR DESCRIPTION
**Pull request by abhaga**

### CODE REVIEW

**Comment by timgraham** [comment]
buildbot, add to whitelist.

File: docs/howto/custom-file-storage.txt [hunk 1]
```diff
@@ -88,8 +88,9 @@ instead).
 .. method:: get_valid_name(name)
 
 Returns a filename suitable for use with the underlying storage system. The
-``name`` argument passed to this method is the original filename sent to the
-server, after having any path information removed. Override this to customize
+``name`` argument passed to this method is either the original filename sent to 
+the server or the filename returned by ``upload_to`` if ``upload_to`` was a 
+callable, after having any path information removed. Override this to customize
```
Comment (timgraham) [inline docs/howto/custom-file-storage.txt]: Should have a ``.. versionchanged:: 1.9`` annotation below this paragraph with a description of the change, as well as a mention in the 1.9 release notes.

**Comment by abhaga** [comment]
@timgraham If something more needs to be done in this PR, please let me know.

**Comment by timgraham** [comment]
merged in 9de9c240178bd9f61f55dad07e76ea458f0291f5, thanks!

**Pull request closed without merging**